src/SelfServe/Documentation/README.md

```diff
@@ -0,0 +1,197 @@
+
+What this means for the feature team 
+- Can concentrate just on the logic behind showing, hiding and disabling UI components 
+- Need not worry about specifics of the UI language
+- Can own the REST API calls made as part of the feature, which may change in the future
+- Quicker turn around time for feature bugs, which they own and have deeper knowledge of
+
+What this means for the UI team
+- Lesser feature development time, since they only own the framework and not the feature itself
+
+Each team owns what they know best, a win-win!
+
+## Getting Started
+
+Clone the cosmos-explorer repo and run
+
+- `npm install`
+- `npm run build`
+
+## Developing
+
+### Watch mode
+
+Run `npm start` to start the development server and automatically rebuild on changes
+
+### Local Development
+
+Update [SelfServeType](../enums/selfserve_selfserveutils.selfservetype.html) to add your feature.
+
+Add the feature flag `feature.dataExplorerSource=https://localhost:1234/selfServe.html?selfServeType%3D<SELF_SERVE_TYPE>` to open up the your feature's UI in the data explorer blade of the portal.
+
+For example, if you want to open up the SelfServeExample UI, visit `https://ms.portal.azure.com/?feature.dataExplorerSource=https://localhost:1234/selfServe.html?selfServeType%3Dexample`
+
+![](https://sdkctlstore.blob.core.windows.net/exe/selfserveDevelopment.PNG)
+
+
+## Code Changes
+
+### 1. Class file
+
+#### Naming Convention
+SELF_SERVE_CLASS_NAME.tsx
+
+#### Example
+[SelfServeExample.tsx](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/Example/SelfServeExample.tsx)
+
+#### Description
+This file will contain the actual code that is transalted into the UI component by the Self Serve framework.
+* Each Self Serve class
+  * Needs to extends the [SelfServeBase](../classes/selfserve_selfservetypes.selfservebaseclass.html) class.
+  * Needs to have the [@IsDisplayable()](./selfserve_decorators.html#isdisplayable) decorator to tell the compiler that UI needs to be generated from this class.
+  * Needs to define an [onSave()](../classes/selfserve_selfservetypes.selfservebaseclass.html#onsave) function, a callback for when the save button is clicked.
+  * Needs to define an [initialize()](../classes/selfserve_selfservetypes.selfservebaseclass.html#initialize) function, to set default values for the inputs.
+  * Needs to define an [onRefresh()](../classes/selfserve_selfservetypes.selfservebaseclass.html#onrefresh) function, a callback for when the refresh button is clicked.
+  * Can have an optional [@RefreshOptions()](./selfserve_decorators.html#refreshoptions) decorator that determines how often the auto refresh of the UI component should take place.
+
+* Each property of the Self Serve class
```

Comment:
>Serve [](http://example.com/codeflow?start=28&length=5)

Each property of the self serve class that needs to be translated into an UI element
a) should have ...Values...
b)..
c)...